packages/subscriptions/__tests__/integration.jest.js

```diff
@@ -78,5 +81,165 @@ describe('subscriptions', () => {
+    const { pgdb } = context
+    const [author, subscriber] = createUsers(2, ['member'])
+    const subscriberContext = { ...context, user: subscriber }
+
+    const subProps = {
+      userId: subscriber.id,
+      objectId: author.id,
+      type: 'User'
+    }
+    const simulatedSubscription = getSimulatedSubscriptionForUserAndObject(
+      subProps.userId,
+      {
+        type: subProps.type,
+        id: subProps.objectId
+      },
+      subscriberContext
+    )
+    const subscription = await upsertSubscription(
+      subProps,
+      subscriberContext
+    )
+    expect(subscription).toBeTruthy()
+    expect(subscription.active).toBeTruthy()
+    expect(subscription.id).toBe(simulatedSubscription.id)
+    expect(await pgdb.public.subscriptions.count()).toBe(1)
+
+    const userSubscriptions = await getSubscriptionsForUser(
+      subProps.userId,
+      subscriberContext
+    )
+    expect(userSubscriptions.length).toBe(1)
+    expect(userSubscriptions[0].id).toBe(subscription.id)
+
+    expect(
+      await getSubscriptionsForUserAndObject(
+        subProps.userId,
+        {
+          type: subProps.type,
+          id: subProps.objectId
+        },
+        subscriberContext
+      ).then( subs => subs?.pop()?.id )
+    ).toBe(subscription.id)
+
+    expect(
+      await getSubscriptionsForUserAndObjects(
+        subProps.userId,
+        {
+          type: subProps.type,
+          ids: [subProps.objectId]
+        },
+        subscriberContext
+      ).then( subs => subs?.pop()?.id )
+    ).toBe(subscription.id)
+
+    await unsubscribe(subscription.id, context)
+    expect(await pgdb.public.subscriptions.count()).toBe(1)
+    expect(await pgdb.public.subscriptions.findOne({
+      id: subscription.id,
+      active: false
+    })).toBeTruthy()
+
+    expect(await getSubscriptionsForUser(
+      subProps.userId,
+      subscriberContext
+    ).then( a => a.length)).toBe(0)
+
+    expect(await getSubscriptionsForUser(
+      subProps.userId,
+      subscriberContext,
+      { includeNotActive: true }
+    ).then( a => a.length)).toBe(1)
+  }),
+
+  test('comments with filters', async () => {
+    const {
+      upsertSubscription,
+      getUnreadNotificationsForUserAndObject
+    } = require('../lib/Subscriptions')
+    const createDiscussion = require('@orbiting/backend-modules-discussions/graphql/resolvers/_mutations/createDiscussion')
+    const submitComment = require('@orbiting/backend-modules-discussions/graphql/resolvers/_mutations/submitComment')
+
+    const { context } = global.instance
+    const { pgdb } = context
+    const [editor] = createUsers(2, ['editor'])
+    const [author, subscriber] = createUsers(2, ['member', 'debater'])
+
+    const filters = [null, 'Comment', 'Document']
+
+    const discussionId = await createDiscussion(
+      null,
+      {
+        title: 'test',
+        anonymity: 'ALLOWED'
+      },
+      { ...context, user: editor }
+    )
+
+    for(let filter of filters) {
```

Comment:
Maybe test in test:

```js
test('comments with filters', () => {
  for(...) {
    test(`filter "${filter}"`, () => {
    
    }
  }
}
```

Reply:
`Tests cannot be nested. Test "filter null" cannot run because it is nested within "comments with filters".`

Reply:
My bad. I didn't read the manual.

Ment this:

```js
describe('age', () => {
  for(...) {
    test('age()', () => {
    }
}
```

Similar to https://github.com/orbiting/backends/blob/master/servers/republik/modules/crowdfundings/lib/Pledge/cancel.jest.js (albeit a unit test here).

Reply:
check it out... that was the only way I could scope the tests. All other methods resulted in strange circular require errors.

Reply:
LGTM. Thanks for the effort.